model/Hierarchy.php

```diff
@@ -373,7 +393,8 @@ public function markedNodeIDs() {
 
 	/**
 	 * Return an array of this page and its ancestors, ordered item -> root.
-	 * @return array
+	 *
+	 * @return SiteTree[]
```

Comment:
As this function literally returns an array of `SiteTree` objects (or their decendants) isn't this correct? Wouldn't being more specific be better, especially for static analysis?  Additionally, with the `SiteTree[]` version an IDE will now be more helpful in situations like:

```php
$page = Page::get()->last();
$parents = $page->parentStack();
$parents[0]-> // IDE will now autocomplete SiteTree properties and methods
```
In cases where an `SS_List` of objects is returned I have always specified the list object rather than fake it as an `array`.